File: crates/klyron_registry/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_registry {
RegistryBuilder::RegistryBuilder() : config_() {}
RegistryBuilder& RegistryBuilder::enabled(bool v) { config_.enabled = v; return *this; }
RegistryBuilder& RegistryBuilder::verbose(bool v) { config_.verbose = v; return *this; }
RegistryConfig RegistryBuilder::build() { return config_; }
}

File: crates/klyron_registry/bindings/rust/builder.rs
use crate::config::RegistryConfig;

pub struct RegistryBuilder {
    config: RegistryConfig,
}

impl RegistryBuilder {
    pub fn new() -> Self { Self { config: RegistryConfig::default() } }
    pub fn build(self) -> RegistryConfig { self.config }
}

impl Default for RegistryBuilder {
    fn default() -> Self { Self::new() }
}
